app/main.py

```diff
@@ -2,0 +1,23 @@
+from app.cafe import Cafe
+from app.errors import VaccineError, NotWearingMaskError
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    count = 0
+    without_mask = 0
+    without_vaccine = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            without_vaccine += 1
+        except NotWearingMaskError:
+            without_mask += 1
+        else:
+            count += 1
+    if count == len(friends):
+        return f"Friends can go to {cafe.name}"
+    if without_vaccine > 0:
+        return "All friends should be vaccinated"
+    if without_vaccine == 0 and without_mask > 0:
+        return f"Friends should buy {without_mask} masks"
```

Comment:
The logic here assumes that if `without_vaccine > 0`, then the return statement will be executed, and the subsequent conditions will not be checked. Ensure that the logic covers all possible scenarios, especially when both vaccine and mask conditions might be violated.